### PR TITLE
[6.x] Fix types in docblocks

### PR DESCRIPTION
**Pull request by iamgergo**
When using `mixed` type in docblocks, it means the type can be basically anything. So, using multiple types like `mixed|array`, `mixed|null` and so on, does not really make sense, because `mixed` already includes any type.

This PR also:
- adds a missing new line in the `Illuminate\Support\Manager` class
- adds missing `@throws` tag
- fixes other doc param and return types

### CODE REVIEW

File: src/Illuminate/Foundation/helpers.php [hunk 1]
```diff
@@ -110,7 +110,7 @@ function action($name, $parameters = [], $absolute = true)
      *
      * @param  string|null  $abstract
      * @param  array  $parameters
-     * @return mixed|\Illuminate\Contracts\Foundation\Application
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/helpers.php]: While I agree with this change, people are going to complain it messes up IDE autocompletion if you remove this.

File: src/Illuminate/Foundation/helpers.php [hunk 2]
```diff
@@ -228,7 +228,7 @@ function broadcast($event = null)
      * If an array is passed, we'll assume you want to put to the cache.
      *
      * @param  dynamic  key|key,default|data,expiration|null
-     * @return mixed|\Illuminate\Cache\CacheManager
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/helpers.php]: Same with all of these.

File: src/Illuminate/Foundation/helpers.php (hunk 1 above)
Comment (iamgergo) [reply src/Illuminate/Foundation/helpers.php]: Okay, I see. Should I put it back then?

Comment (GrahamCampbell) [reply src/Illuminate/Foundation/helpers.php]: Yes, I think so.

**Comment by lucasmichot** [comment]
Is there a reason why many `array|mixed` docblocks remain in the code?

Like for instance `\Illuminate\Cache\RedisStore::tags`, or many in `\Illuminate\Auth\Access\Gate`
But not only.

**Comment by iamgergo** [comment]
Yes, I was searching only for `mixed|*`... I'll fix them, thank you.

File: src/Illuminate/Support/Facades/Request.php [hunk 3]
```diff
@@ -62,7 +62,7 @@
  * @method static string userAgent()
  * @method static \Illuminate\Http\Request merge(array $input)
  * @method static \Illuminate\Http\Request replace(array $input)
- * @method static \Symfony\Component\HttpFoundation\ParameterBag|mixed json(string|null $key = null, mixed $default = null)
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Facades/Request.php]: Please revert this change.

File: src/Illuminate/Support/LazyCollection.php [hunk 4]
```diff
@@ -812,7 +812,7 @@ public function concat($source)
      * Get one or a specified number of items randomly from the collection.
      *
      * @param  int|null  $number
-     * @return static|mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/LazyCollection.php]: Please revert this change.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
Some more changes to be reverted please, since they are there to help IDEs.

File: src/Illuminate/Support/Facades/Request.php (hunk 3 above)
Comment (iamgergo) [reply src/Illuminate/Support/Facades/Request.php]: I changed other facades as well. You want them back as well?

Comment (GrahamCampbell) [reply src/Illuminate/Support/Facades/Request.php]: Any return types that used to have an object combined with mixed, should be retained.

Comment (iamgergo) [reply src/Illuminate/Support/Facades/Request.php]: Reverted all **return types** where object was combined with mixed.

**Comment by taylorotwell** [comment]
This will need to go to master. I don't like huge PR's like this on the current branch.

**Comment by iamgergo** [comment]
Okay, I'll make a PR into master.
